website/client/components/chat/chatMessages.vue

```diff
@@ -27,7 +27,7 @@
           span.action( @click='copyAsTodo(msg)')
             .svg-icon(v-html="icons.copy")
             | {{$t('copyAsTodo')}}
-          span.action(v-if='user.contributor.admin || (!msg.sent && user.flags.communityGuidelinesAccepted)', @click='report(msg)')
+          span.action(v-if='user.contributor.admin || (msg.uuid !== user._id && user.flags.communityGuidelinesAccepted)', @click='report(msg)')
```

Comment:
In prod you can flag your own messages (even if you aren't a mod). I might have misread your code - does it stop a player flagging their own message?